File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testdata/sync/CacheableTargetSync.java
package ru.tinkoff.kora.cache.annotation.processor.testdata.sync;

import ru.tinkoff.kora.cache.annotation.CacheInvalidate;
import ru.tinkoff.kora.cache.annotation.CachePut;
import ru.tinkoff.kora.cache.annotation.Cacheable;
import ru.tinkoff.kora.cache.annotation.processor.testcache.DummyCacheManager;

import java.math.BigDecimal;

public class CacheableTargetSync {

    public String value = "1";

    @Cacheable(name = "sync_cache", tags = DummyCacheManager.class)
    public String getValue(String arg1, BigDecimal arg2) {
        return value;
    }

    @CachePut(name = "sync_cache", tags = DummyCacheManager.class, parameters = {"arg1", "arg2"})
    public String putValue(BigDecimal arg2, String arg3, String arg1) {
        return value;
    }

    @CacheInvalidate(name = "sync_cache", tags = DummyCacheManager.class)
    public void evictValue(String arg1, BigDecimal arg2) {

    }

    @CacheInvalidate(name = "sync_cache", tags = DummyCacheManager.class, invalidateAll = true)
    public void evictAll() {

    }
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testdata/CacheableTargetSync.kt
package ru.tinkoff.kora.cache.symbol.processor.testdata

import ru.tinkoff.kora.cache.annotation.CacheInvalidate
import ru.tinkoff.kora.cache.annotation.CachePut
import ru.tinkoff.kora.cache.annotation.Cacheable
import ru.tinkoff.kora.cache.symbol.processor.testcache.DummyCacheManager
import java.math.BigDecimal

open class CacheableTargetSync {
    var value = "1"

    @Cacheable(name = "sync_cache", tags = [DummyCacheManager::class])
    open fun getValue(arg1: String?, arg2: BigDecimal?): String {
        return value
    }

    @CachePut(name = "sync_cache", tags = [DummyCacheManager::class], parameters = ["arg1", "arg2"])
    open fun putValue(arg2: BigDecimal?, arg3: String?, arg1: String?): String {
        return value
    }

    @CacheInvalidate(name = "sync_cache", tags = [DummyCacheManager::class])
    open fun evictValue(arg1: String?, arg2: BigDecimal?) {
    }

    @CacheInvalidate(name = "sync_cache", tags = [DummyCacheManager::class], invalidateAll = true)
    open fun evictAll() {
    }
}
